CIP-0062/README.md

```diff
@@ -0,0 +1,220 @@
+---
+CIP: 62
+Title: Cardano dApp-Connector Governance extension
+Authors: Bruno Martins <bruno.martins@iohk.io>
+Status: Draft
+Type: Standards
+Created: 2021-06-11
+License: CC-BY-4.0
+---
+
+# **Abstract**
+
+This document describe the interface between webpage / web-based stack and cardano wallets. This specificies that API of the javascript object that need to be injected into the web applications in order to support all the Governance features.
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# **Motivation**
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# **Specification**
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+
+```
+
+`votingKey`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
```

Comment:
Suggestion: reverse the order here. Keep `0` for other / misc, `1` for Catalyst and leave the door open for new purposes to be added after 1 :+1:

Reply:
This reflects CIP-36.  0 was chosen for backwards compatibility with CIP-15.  The intention is indeed to use purposes from 1 onwards for various voting situations, in the same way that there is eg an RFC for port numbers, and a different RFC that defines the conventional uses of various ports.  At this point in time, we don't know what those purposes will be

Reply:
I think `OTHER = 1` should be removed from the enum, its confusing as its not a real or defined current "Purpose".
There should just be a comment that says, other purposes may be defined in future revisions to this CIP.

Reply:
I'd suggest finding a better type name, more self-explanatory than `Purpose.`